messaging/src/main/java/org/axonframework/deadline/dbscheduler/DbSchedulerDeadlineManager.java

```diff
@@ -0,0 +1,375 @@
+        runOnPrepareCommitOrNow(span.wrapRunnable(
+                () -> scheduler.cancel(new DbSchedulerDeadlineToken(scheduleId)))
+        );
+    }
+
+    @Override
+    public void cancelAll(@Nonnull String deadlineName) {
+        Span span = spanFactory.createInternalSpan(() -> "DbSchedulerDeadlineManager.cancelAll(" + deadlineName + ")");
+        runOnPrepareCommitOrNow(span.wrapRunnable(
+                () -> scheduler.fetchScheduledExecutionsForTask(
+                        TASK_NAME,
+                        DbSchedulerDeadlineDetails.class,
+                        cancelIfDeadlineMatches(deadlineName)
+                )));
+    }
+
+    private Consumer<ScheduledExecution<DbSchedulerDeadlineDetails>> cancelIfDeadlineMatches(
+            @Nonnull String deadlineName
+    ) {
+        return scheduledExecution -> {
+            if (deadlineName.equals(scheduledExecution.getData().getDeadlineName())) {
+                scheduler.cancel(scheduledExecution.getTaskInstance());
+            }
+        };
+    }
+
+    @Override
+    public void cancelAllWithinScope(@Nonnull String deadlineName, @Nonnull ScopeDescriptor scope) {
+        Span span = spanFactory.createInternalSpan(
+                () -> "DbSchedulerDeadlineManager.cancelAllWithinScope(" + deadlineName + ")"
+        );
+        runOnPrepareCommitOrNow(span.wrapRunnable(
+                () -> {
+                    SerializedObject<String> serializedDescriptor = serializer.serialize(scope, String.class);
+                    scheduler.fetchScheduledExecutionsForTask(
+                            TASK_NAME,
+                            DbSchedulerDeadlineDetails.class,
+                            cancelIfDeadlineAndScopeMatches(deadlineName, serializedDescriptor.getData()));
+                }));
+    }
+
+    private Consumer<ScheduledExecution<DbSchedulerDeadlineDetails>> cancelIfDeadlineAndScopeMatches(
+            @Nonnull String deadlineName,
+            @Nonnull String scopeDescriptor
+    ) {
+        return scheduledExecution -> {
+            DbSchedulerDeadlineDetails data = scheduledExecution.getData();
+            if (deadlineName.equals(data.getDeadlineName()) && scopeDescriptor.equals(data.getScopeDescriptor())) {
+                scheduler.cancel(scheduledExecution.getTaskInstance());
+            }
+        };
+    }
+
+    /**
+     * This function is used by the {@link #task()} to execute the deadline.
+     *
+     * @param deadlineDetails {@link DbSchedulerDeadlineDetails} containing the needed details to execute.
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private void execute(DbSchedulerDeadlineDetails deadlineDetails) {
+        GenericDeadlineMessage deadlineMessage = deadlineDetails.asDeadLineMessage(serializer);
+        Span span = spanFactory.createLinkedHandlerSpan(() -> "DeadlineJob.execute", deadlineMessage).start();
+        try (SpanScope unused = span.makeCurrent()) {
```

Comment:
The span is not ended in a finally block

Reply:
fixed, also fo the `JobRunrDeadlineManager`.